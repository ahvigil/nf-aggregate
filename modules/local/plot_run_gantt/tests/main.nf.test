nextflow_process {

    name "Test Process PLOT_RUN_GANTT"
    script "../main.nf"
    process "PLOT_RUN_GANTT"
    config "./nextflow.config"

    tag "modules"
    tag "modules_local"
    tag "plot_run_gantt"

    test("Should run without failures") {

        setup {
            run("SEQERA_RUNS_DUMP") {
                script "../../seqera_runs_dump/main.nf"
                process {
                    """
                    input[0] = ['id': '4Bi5xBK6E2Nbhj', 'workspace': 'community/showcase']
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = SEQERA_RUNS_DUMP.out.run_dump.join(SEQERA_RUNS_DUMP.out.workflow_json).map { meta, run_dump, json -> [ meta + [ pipeline : getWorkflowName(json) ], run_dump ] } 
		"""
            }
        }

        then {
            assertAll(
                { assert snapshot(process.out.html).match() },
                { assert process.success }
            )
        }

    }

}
