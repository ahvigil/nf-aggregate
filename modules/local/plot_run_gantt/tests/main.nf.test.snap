{
    "Should run without failures": {
        "content": [
            [
                
            ]
        ],
        "timestamp": "2023-11-28T16:52:51.371307827"
    }
}